src/jit/codegenlinear.cpp

```diff
@@ -1499,11 +1499,12 @@ void CodeGen::genPutArgStkFieldList(GenTreePutArgStk* putArgStk, unsigned outArg
         emitAttr  attr = emitTypeSize(type);
 
         // Emit store instructions to store the registers produced by the GT_FIELD_LIST into the outgoing
-        // argument area
-        getEmitter()->emitIns_S_R(ins_Store(type), attr, reg, outArgVarNum, argOffset);
-        argOffset += EA_SIZE_IN_BYTES(attr);
+        // argument area.
+        unsigned thisArgOffset = argOffset + fieldListPtr->gtFieldOffset;
```

Comment:
I'm not exactly familiar with this code but I'm pretty sure that here "this" does not refer to the `this` argument. Perhaps `argFieldOffset` would be a better name? `argOffset` is the offset of the an argument composed from multiple fields and `argFieldOffset` is the offset of an individual field.

Reply:
Changed

Reply:
Did this fix get pushed? Github says the local is still named `thisFieldOffset`? 

I like `argFieldOffset` here better too.